File: client/ios/Contract/Support/SupportMobilePayload.swift
import Foundation

public struct SupportOptionPayload: Identifiable, Sendable {
    public var id: String { value }
    public let label: String
    public let value: String
}

public struct SupportFieldPayload: Identifiable, Sendable {
    public var id: String { name }
    public let name: String
    public let label: String
    public let type: String
    public let value: String?
    public let required: Bool
    public let options: [SupportOptionPayload]
}

public struct SupportActionPayload: Identifiable, Sendable {
    public var id: String { "\(method):\(href):\(label)" }
    public let label: String
    public let href: String
    public let method: String
    public let enabled: Bool
}

public struct SupportRowPayload: Identifiable, Sendable {
    public let id: String
    public let context: String
    public let request: String
    public let description: String
    public let href: String
    public let availableItems: Int
}

public struct CaseRowPayload: Identifiable, Sendable {
    public var id: String { reference }
    public let reference: String
    public let context: String
    public let category: String
    public let status: String
    public let href: String
}

public struct SupportPagePayload: Sendable {
    public let title: String
    public let description: String
    public let rows: [SupportRowPayload]
    public let cases: [CaseRowPayload]
    public let fields: [SupportFieldPayload]
    public let actions: [SupportActionPayload]
    public let action: String?
    public let method: String
    public let reference: String?
    public let status: String?
    public let businessContext: String?
    public let category: String?
    public let descriptionText: String?
}

File: client/android/Contract/Support/SupportMobilePayload.kt
package app.mobiling.client.contract.support

data class SupportOptionPayload(val label: String, val value: String)

data class SupportFieldPayload(
    val name: String,
    val label: String,
    val type: String,
    val value: String?,
    val required: Boolean,
    val options: List<SupportOptionPayload>,
)

data class SupportActionPayload(
    val label: String,
    val href: String,
    val method: String,
    val enabled: Boolean,
)

data class SupportRowPayload(
    val id: String,
    val context: String,
    val request: String,
    val description: String,
    val href: String,
    val availableItems: Int,
)

data class CaseRowPayload(
    val reference: String,
    val context: String,
    val category: String,
    val status: String,
    val href: String,
)

data class SupportPagePayload(
    val title: String,
    val description: String,
    val rows: List<SupportRowPayload> = emptyList(),
    val cases: List<CaseRowPayload> = emptyList(),
    val fields: List<SupportFieldPayload> = emptyList(),
    val actions: List<SupportActionPayload> = emptyList(),
    val action: String? = null,
    val method: String = "GET",
    val reference: String? = null,
    val status: String? = null,
    val businessContext: String? = null,
    val category: String? = null,
    val descriptionText: String? = null,
)
